src/Paginator.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+namespace WP2Static;
+
+class Paginator {
+    /**
+     * @var array<mixed>
+     */
+    public $records = [];
+
+    /**
+     * @var int
+     */
+    public $page_size = 100;
+
+    /**
+     * @var int
+     */
+    public $page = 1;
+
+    /**
+     * @var int
+     */
+    public $pages = 1;
+
+    /**
+     * @var int
+     */
+    public $total_records = 0;
+
+    /**
+     * Create a new Paginator
+     *
+     * @param array<mixed> $records
+     * @param integer $page_size
+     * @param integer $page
+     */
+    public function __construct( array $records, int $page_size, int $page ) {
+        $this->page = $page;
+        $this->page_size = $page_size;
+        $this->pages = intval( ceil( count( $records ) / $page_size ) );
+        $this->total_records = count( $records );
+        $this->records = $this->paginateRecords( $records, $page_size, $page );
+    }
+
+    public function page() : int {
+        return $this->page;
+    }
+
+    public function nextPage() : int {
+        return min( $this->page + 1, $this->pages );
+    }
+
+    public function prevPage() : int {
+        return max( 1, $this->page - 1 );
+    }
+
+    public function firstPage() : int {
+        return 1;
+    }
+
+    public function lastPage() : int {
+        return $this->pages;
+    }
+
+    public function pageSize() : int {
+        return count( $this->records );
+    }
+
+    public function totalRecords() : int {
+        return $this->total_records;
+    }
+
+    /**
+     * Returns the paginated set of records.
+     *
+     * @return array<mixed>
+     */
+    public function records() : array {
+        return $this->records;
+    }
+
+    public function render() : void {
+        require __DIR__ . '/../views/_paginator.php';
```

Comment:
Can we use the `WP2STATIC_PATH` from https://github.com/WP2Static/wp2static/blob/741c1568ceb000adc019b7d80b2f26580d935a7f/wp2static.php#L18 here?